github/repos_contents.go

```diff
@@ -119,6 +120,10 @@ func (s *RepositoriesService) DownloadContents(owner, repo, filepath string, opt
 	return nil, fmt.Errorf("No file named %s found in %s", filename, dir)
 }
 
+func escapeURLPath(path string) string {
+	return template.URLQueryEscaper(path)
```

Comment:
Updating the test is certainly good, though I was actually referring to testing how the GitHub API actually behaves.  I'll try creating a test repo to verify.

(the two folks I cc'ed above work on the API)